src/main/java/seedu/address/model/cheese/Cheese.java

```diff
@@ -71,8 +73,12 @@ public boolean isSameType(CheeseType cheeseType) {
         return this.cheeseType.equals(cheeseType);
     }
 
+    /**
+     * Returns a cheese with same fields that is marked assigned.
+     */
     public Cheese assignToOrder() {
-        return new Cheese(cheeseType, manufactureDate, maturityDate, expiryDate, cheeseId);
+        return new Cheese(cheeseType, manufactureDate, maturityDate.orElse(null),
+                expiryDate.orElse(null), cheeseId);
     }
 
     /**
```

Comment:
Do update the `toString` method for cheese (now showing `optional.empty`)

Reply:
Yeap this is updated.

Reply:
Yeap this is updated in the latest push.